content-src/components/TopSites/TopSites.js

```diff
@@ -26,22 +26,42 @@ const TopSitesItem = React.createClass({
     const site = this.props;
     const index = site.index;
     const isActive = this.state.showContextMenu && this.state.activeTile === index;
-    const screenshot = site.screenshot;
+
+    const topSitesExperimentIsOn = this.props.showNewStyle;
+    const screenshot = topSitesExperimentIsOn && site.screenshot;
 
     // The top-corner class puts the site icon in the top corner, overlayed over the screenshot.
     const siteIconClasses = classNames("tile-img-container", {"top-corner": screenshot});
 
+    let faviconSize = 32;
+    if (topSitesExperimentIsOn && !screenshot) {
+      if (!site.favicon_width) {
+        // default to 64 if not specified (*cough* tippy top)
+        faviconSize = 64;
+      } else {
+        faviconSize = site.favicon_width;
+      }
+
+      // we want the favicon to be at least 32x32 and at most 96x96
+      if (faviconSize > 96) {
+        faviconSize = 96;
+      } else if (faviconSize < 32) {
+        faviconSize = 32;
+      }
+    }
+
```

Comment:
The render method is pretty giant.  Move the favicon size computations to another method on the same object for readability?